stan/math/prim/scal/meta/partials_return_type.hpp

```diff
@@ -7,16 +7,19 @@
 
 namespace stan {
 
```

Comment:
Needs documentation.

@dpsimpson, I'm not attributing the lack of doc to you. This is just some of the technical debt we've accumulated over the years. I'm also not forcing you to write doc. If you're happy to, then please, by all means. If you don't want to, no worries; we can find a volunteer to write this doc.

Reply:
Documented. Which is good because I actually misunderstood what the return type was and made the unit test incorrectly. I now check that it sends `var` -> `double` and `fvar<var>` -> `double`.

Do I need to test that it sends eg `fvar<float>` -> `double`?  I think this is implied by the other tests.

It would be good to add "when this should be used" to the doc, because it's mainly popping up in modern code.

I'm also not 100% sure that the use of it is always safe for higher-order partials. For example, I'm not sure what lines 93-98 of @VMatthijs `normal_id_glm_lpdf` code. https://github.com/VMatthijs/math/blob/f9d380d7a65ba4978753164c4bf11dcd08dc6f7e/stan/math/prim/mat/prob/normal_id_glm_lpdf.hpp

In that case, if there is a `fvar<var>` in the template parameters, those lines will construct a column vector of `var`s, which are then assigned like doubles. Will this work?  If not, I'll make an issue.

Reply:
Hmm... is that right? I thought `partials_return_type` was the type of the partial itself.

For a `stan::math::var`, the partial should be `double`.

For a `stan::math::fvar<double>`, it should be `double`. (I think if it was `fvar<float>`, it should return `double`, but we don't need to check this at every level of nesting.)

For a `stan::math::fvar<var>`, isn't it `var`? Or am I missing something? The directional derivative returned is a `var` (so we can autodiff through that).

Reply:
Yep. I got it wrong but it's right in the current version. (Misleading result due to having the wrong set of `#include`s.